crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -385,9 +385,14 @@ def prepare_nodes_for_os_upgrade
     upgrade_nodes.each do |node|
       node["target_platform"] = admin_node["provisioner"]["default_os"]
       node["crowbar_wall"]["crowbar_upgrade_step"] = "prepare-os-upgrade"
+      # skip initial keystone bootstrapping
+      node["keystone"]["bootstrap"] = true if node["keystone"]
```

Comment:
Hm, somehow it looks that the change to `node.roles` broke it... I need to investigate more...